JavaAssignments/src/farhin/MissingNoLinear.java

```diff
@@ -0,0 +1,44 @@
+package farhin;
+
+import java.util.Scanner;
+
+public class MissingNoLinear {
+
+	int[] takeInputs() {
+		Scanner sc = new Scanner(System.in);
+		System.out.println("Enter array size : ");
+		int arrSize = sc.nextInt();
+		int[] arrNum = new int[arrSize];
+		for (int index = 0; index < arrSize-1; index++) // Taking numbers in array
+		{
+			arrNum[index] = sc.nextInt();
+			
+		}
+		sc.close();
+		return arrNum;
+	}
+
+	int calcActualSum(int[] numArr) {
+		int actualSum = 0;
+		for (int index = 0; index < numArr.length; index++)  //to calculate actual sum of array
+		{
+			actualSum = actualSum + numArr[index];
+		}
+		
+		return actualSum;
+	}
+
+	int missingNum(int actualSum) {
+		int expectedSum = 55;
```

Comment:
how can we hardcode expected sum ??
if user asked to find missing number between 1 to 100 then ??
you should calculate expected sum = n
*(n+1)/2

Reply:
I do understand you wrote it for 1->10 only but lets change it .